### PR TITLE
[docs] Apply style guide to Data Grid Aggregation page

### PR DESCRIPTION
**Pull request by samuelsycamore**
Part of #5764 

This PR revises the Data Grid Aggregation page, addressing style, grammar, and formatting. This page was created recently so it was already in pretty good shape. This edit mainly trims the fat, replaces passive voice and future tense, and applies consistent styles to names of UI elements.

### CODE REVIEW

**Review by flaviendelangle — APPROVED** [review]

**Review by flaviendelangle — APPROVED** [review]

File: docs/data/data-grid/aggregation/aggregation.md [hunk 1]
```diff
@@ -2,16 +2,17 @@
 title: Data Grid - Aggregation
 ---
 
-# Data grid - Aggregation [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan)
+# Data Grid - Aggregation [<span class="plan-premium"></span>](https://mui.com/store/items/mui-x-premium/)
```
Comment (oliviertassinari) [inline docs/data/data-grid/aggregation/aggregation.md:5]:
```suggestion
# Data Grid - Aggregation [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan)
```

File: docs/data/data-grid/aggregation/aggregation.md [hunk 2]
```diff
@@ -36,49 +37,44 @@ To initialize aggregation without controlling its state, provide the model to th
 
 ### Controlled aggregation
 
-Use the `aggregationModel` prop to control aggregation passed to the grid.
+Use the `aggregationModel` prop to control aggregation passed to the Data Grid.
```
Comment (oliviertassinari) [inline docs/data/data-grid/aggregation/aggregation.md:40]: It's not a header here:

```suggestion
Use the `aggregationModel` prop to control aggregation passed to the data grid.
```

File: docs/data/data-grid/aggregation/aggregation.md [hunk 3]
```diff
@@ -215,16 +208,16 @@ const firstAlphabeticalAggregation: GridAggregationFunction<string, string | nul
   };
 ```
 
-In the example below, the grid has two additional custom aggregation functions for `string` columns—`firstAlphabetical` and `lastAlphabetical`:
+In the example below, the Data Grid has two additional custom aggregation functions for `string` columns: `firstAlphabetical` and `lastAlphabetical`:
```
Comment (oliviertassinari) [inline docs/data/data-grid/aggregation/aggregation.md:211]:
```suggestion
In the example below, the data grid has two additional custom aggregation functions for `string` columns: `firstAlphabetical` and `lastAlphabetical`:
```

**Review by joserodolfofreitas — APPROVED** [review]
Thanks for the updates, @samuelsycamore!

**Comment by samuelsycamore** [comment]
Something very weird happened to this PR. 😅 I have no clue what's going on!

**Pull request merged**